go/controller/internal/a2a/autogen_a2a_translator.go

```diff
@@ -99,11 +99,15 @@ func (a *autogenA2ATranslator) makeHandlerForTeam(
 	return func(ctx context.Context, task string, sessionID *string) (string, error) {
 		var taskResult *autogen_client.TaskResult
 		if sessionID != nil && *sessionID != "" {
-			session, err := a.autogenClient.GetSession(*sessionID, common.GetGlobalUserID())
+			sessionName := fmt.Sprintf("%v-%s-%s", autogenTeam.CreatedAt, autogenTeam.UserID, *sessionID)
+			log.FromContext(ctx).Info("Using session for task", "sessionName", sessionName)
+			session, err := a.autogenClient.GetSession(sessionName, common.GetGlobalUserID())
```

Comment:
I'm not so sure about this change, why aren't we just using the `sessionID` directly?